java/src/test/java/ai/rapids/cudf/ast/CompiledExpressionTest.java

```diff
@@ -476,6 +476,65 @@ void testBinaryComparisonOperationTransform(BinaryOperator op, Integer[] in1, In
     }
   }
 
+  private static Stream<Arguments> createStringLiteralComparisonParams() {
+    String[] in1 = new String[] {"a", "bb", null, "ccc", "dddd"};
+    String in2 = "ccc";
+    return Stream.of(
+        // nulls compare as equal by default
+        Arguments.of(BinaryOperator.NULL_EQUAL, in1, in2, Arrays.asList(false, false, false, true, false)),
+        Arguments.of(BinaryOperator.NOT_EQUAL, in1, in2, mapArray(in1, (a) -> !a.equals(in2))),
+        Arguments.of(BinaryOperator.LESS, in1, in2, mapArray(in1, (a) -> a.compareTo(in2) < 0)),
+        Arguments.of(BinaryOperator.GREATER, in1, in2, mapArray(in1, (a) -> a.compareTo(in2) > 0)),
+        Arguments.of(BinaryOperator.LESS_EQUAL, in1, in2, mapArray(in1, (a) -> a.compareTo(in2) <= 0)),
+        Arguments.of(BinaryOperator.GREATER_EQUAL, in1, in2, mapArray(in1, (a) -> a.compareTo(in2) >= 0)));
+  }
+
+  @ParameterizedTest
+  @MethodSource("createStringLiteralComparisonParams")
+  void testStringLiteralComparison(BinaryOperator op, String[] in1, String in2,
+                                               List<Boolean> expectedValues) {
+    Literal lit = Literal.ofString(in2);
```

Comment:
This doesn't test a null string literal, should add that as a test case.

I also recommend adding something like the `test*LiteralTransform` tests which are more focused on just handling the literal directly without other expressions and validates that the string data was translated properly.  The binary comparison test can hide a bad string translation and still pass the test by chance.

Reply:
For string column, direct transformation is not supported. only fixed width types are supported now.